librz/core/rtr_http.c

```diff
@@ -1,562 +0,0 @@
-
-		if (!rs) {
-			bed = rz_cons_sleep_begin ();
-			rz_sys_usleep (100);
-			rz_cons_sleep_end (bed);
-			continue;
-		}
-		if (allow && *allow) {
-			bool accepted = false;
-			const char *allows_host;
-			char *p, *peer = rz_socket_to_string (rs->s);
-			char *allows = strdup (allow);
-			//eprintf ("Firewall (%s)\n", allows);
-			int i, count = rz_str_split (allows, ',');
-			p = strchr (peer, ':');
-			if (p) {
-				*p = 0;
-			}
-			for (i = 0; i < count; i++) {
-				allows_host = rz_str_word_get0 (allows, i);
-				//eprintf ("--- (%s) (%s)\n", host, peer);
-				if (!strcmp (allows_host, peer)) {
-					accepted = true;
-					break;
-				}
-			}
-			free (peer);
-			free (allows);
-			if (!accepted) {
-				rz_socket_http_close (rs);
-				continue;
-			}
-		}
-		if (!rs->method || !rs->path) {
-			http_logf (core, "Invalid http headers received from client\n");
-			rz_socket_http_close (rs);
-			continue;
-		}
-		dir = NULL;
-
-		if (!rs->auth) {
-			rz_socket_http_response (rs, 401, "", 0, NULL);
-		}
-
-		if (rz_config_get_i (core->config, "http.verbose")) {
-			char *peer = rz_socket_to_string (rs->s);
-			http_logf (core, "[HTTP] %s %s\n", peer, rs->path);
-			free (peer);
-		}
-		if (rz_config_get_i (core->config, "http.dirlist")) {
-			if (rz_file_is_directory (rs->path)) {
-				dir = strdup (rs->path);
-			}
-		}
-		if (rz_config_get_i (core->config, "http.cors")) {
-			strcpy (headers, "Access-Control-Allow-Origin: *\n"
-				"Access-Control-Allow-Headers: Origin, "
-				"X-Requested-With, Content-Type, Accept\n");
-		}
-		if (!strcmp (rs->method, "OPTIONS")) {
-			rz_socket_http_response (rs, 200, "", 0, headers);
-		} else if (!strcmp (rs->method, "GET")) {
-			if (!strncmp (rs->path, "/up/", 4)) {
-				if (rz_config_get_i (core->config, "http.upget")) {
-					const char *uproot = rz_config_get (core->config, "http.uproot");
-					if (!rs->path[3] || (rs->path[3]=='/' && !rs->path[4])) {
-						char *ptr = rtr_dir_files (uproot);
-						rz_socket_http_response (rs, 200, ptr, 0, headers);
-						free (ptr);
-					} else {
-						char *path = rz_file_root (uproot, rs->path + 4);
-						if (rz_file_exists (path)) {
-							size_t sz = 0;
-							char *f = rz_file_slurp (path, &sz);
-							if (f) {
-								rz_socket_http_response (rs, 200, f, (int)sz, headers);
-								free (f);
-							} else {
-								rz_socket_http_response (rs, 403, "Permission denied", 0, headers);
-								http_logf (core, "http: Cannot open '%s'\n", path);
-							}
-						} else {
-							if (dir) {
-								char *resp = rtr_dir_files (dir);
-								rz_socket_http_response (rs, 404, resp, 0, headers);
-								free (resp);
-							} else {
-								http_logf (core, "File '%s' not found\n", path);
-								rz_socket_http_response (rs, 404, "File not found\n", 0, headers);
-							}
-						}
-						free (path);
-					}
-				} else {
-					rz_socket_http_response (rs, 403, "", 0, NULL);
-				}
-			} else if (!strncmp (rs->path, "/cmd/", 5)) {
```

Comment:
This does break rz-pipe with `http://`. Removal of http while still keeping rz-pipe intact would mean only removing the file-serving but keeping the cmd stuff.
We might want to reconsider keeping http rz-pipe since there is also the tcp one but imo this should be discussed.